### ISSUE TITLE
general discussion

### ISSUE THREAD

**Issue by myrdd** [RequestPolicyContinued/requestpolicy#484]
This issue is for general discussion and any kind of short comments or questions. Anything that doesn't need a separate issue can be written here. Off-topic allowed.

other useful discussions:

* documentation (and wiki) discussion: https://github.com/RequestPolicyContinued/requestpolicy/issues/902
* developers' discussion: https://github.com/RequestPolicyContinued/requestpolicy/issues/826

**Comment by nodiscc** [comment]
I've read the Disconnect addon (mentioned in https://github.com/RequestPolicyContinued/requestpolicy/issues/448#issuecomment-53158340) is [now proprietary](https://github.com/nylira/prism-break/issues/1214#issuecomment-69272807). I used to recommend/install disconnect for people who couldn't handle RP in _default: block_, but if we can

1) get blocklists from Disconnect into RPcontinued subscriptions
2) Have a simple first run page (https://github.com/RequestPolicyContinued/requestpolicy/issues/545) proposing _default block/allow functionality_ vs _default allow/block trackers_ modes

this would easily replace disconnect (and maybe ghostery)

Updating the website https://github.com/RequestPolicyContinued/requestpolicy/issues/445 is still on my list :/

**Comment by myrdd** [comment]
Disconnect is now proprietary? Hm, too bad! I've also installed it on some machines…

> get blocklists from Disconnect into RPcontinued subscriptions

You want to do this? I'm focussing heavily on coding right now. RP 0.5 is marked as unstable with Fx 36, wich will be released in the late February…

> Have a simple first run page (#545) proposing default block/allow functionality vs default allow/block trackers modes

Yes, I also consider a good initial setup page as important – however, my idea is to put the decision between `default allow vs. default deny` prominently into the menu, either in addition or instead of an "initial setup" page.

> Updating the website #445 is still on my list :/

No worries, as long as the content of the current website is mostly up to date, I see no problem.

**Comment by nodiscc** [comment]
After looking at [µBlock](https://github.com/gorhill/uBlock) (a lightweight ad blocker), I think it implements some form of "local mirroring" of common js/css/font resources: https://github.com/gorhill/uBlock/blob/master/assets/ublock/mirror-candidates.txt. It seems to have been removed in a recent release though: https://github.com/gorhill/uBlock/releases 

This (mirroring resources from common CDNs) was discussed at https://github.com/RequestPolicyContinued/requestpolicy/issues/267 for RequestPolicy

**Comment by myrdd** [comment]
That's interesting, I think it's a nice feature.

**Comment by rezad1393** [comment]
hi
i use both your addon and policeman addon together so that the requests get filtered first by RP addon then the allowed one get filtered by policeman addon. today when i was playing with RP(requestpolicy) addon policies something strange happened.from that moment on the requests go to policeman addon first then the allowed ones go to RP addon.but this was reverse before i somehow messed it up.

how can i fix this(reverse to old behavior?)

**Comment by myrdd** [comment]
Hi @rezad1393 if I remember correctly, the order is determined by the order of installation, so you couly try reinstalling the addons in the order you want, maybe that solves the problem?

Ah yes, I've read it there: https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIContentPolicy#shouldLoad%28%29

>  Note: When multiple content policies are used (for example through several extensions), if one of them rejects a request, the rest of the policies are not called. Reference: http://mxr.mozilla.org/mozilla-central/source/content/base/src/nsContentPolicy.cpp#146.

>  Note: The order of content policies in the above case depends on the precedence of installation.

**Comment by rezad1393** [comment]
thanks man
you were correct

**Comment by rezad1393** [comment]
@myrdd it seems I reported too early.
the workaround you said works at first but after a while it goes back to the reserve way(policeman first then RP addon)

what do I do?
can I specify the order of addons that work on requests outside of installation order? cause that didn't last.

**Comment by myrdd** [comment]
@rezad1393 I could imagine that each time an addon gets updated it gets back to the top of the list. I guess this needs to be tested. Please open an issue, then I'll investigate when I have time for it. In the end I guess RP can't change anything at Firefox' behaviour; maybe this issue has been discussed in a Mozilla Bug – https://bugzilla.mozilla.org/

**Comment by rezad1393** [comment]
maybe i didnt say this right.English is not my first language.
i dont change these two addons after temporary fixes of uninstalling and reinstalling the policeman.
but after using firefox for a while it goes to the the policaman->requestpolicy  again.

this happens both 35 and 36 versions of firefox.
i dont think its your addon bug (at least not in design) and i dont know what i did that made this happen.
i just remember that it was after changing some of the requestpolicy setting.

**Comment by myrdd** [comment]
Hi @rezad1393 ! To discuss this further, please open a new issue. It would be good to find out how to reproduce your issue. You could test whether you have the same problem when using a fresh new profile.

> uninstalling and reinstalling the policeman

I found out that disabling and enabling policeman does the same regarding `shouldLoad`. This is at least a faster workaround.

**Comment by rezad1393** [comment]
ok created an issue ticket here 
https://github.com/RequestPolicyContinued/requestpolicy/issues/603
and as i said there i created a  new profile and just imported my bookmark and history and installed my enabled addons from old profile but that didnt fix it.
i resume the discussion there if somebody is able to help me.

**Comment by shirishag75** [comment]
First question, how does one create screenshot of the Request Policy as can be seen in https://github.com/RequestPolicyContinued/requestpolicy/issues/484#issuecomment-57916812 .  I have an issue which I would like to explain using the screenshot for easier conversation.

I am on Debian GNU/Linux.

**Comment by rezad1393** [comment]
#shirishag75 
this are rules added by the addon itself that are called subscriptions.
they are not added by you manually. think of them as groups of rules that serve a specific purpose. like in your provided example that is for blocking advertisement on imgur website.
more info here

https://github.com/RequestPolicyContinued/subscriptions

**Comment by shirishag75** [comment]
@rezad1393 thank you. That makes sense BUT this now puts me in a fix. I have a site (meta.unix.stackexchange.com) where the imgur.com is there in allowed destinations but with a question mark and I have no idea why the question mark is there.  

Now that issue seems to be related to this issue which I raised on https://meta.stackexchange.com/questions/250599/unable-to-login-at-times-on-meta-unix-stackexchange-com which takes me to https://meta.stackexchange.com/questions/56161/which-browsers-are-officially-supported-and-what-else-do-i-need/71208#71208

Out of them, most of them work except for two, the first I have shared about imgur and the second one is the automated one about network logging at https://meta.stackoverflow.com/network-login-help :-
   
Messaging failed

Your browser does not support the HTML5 postMessage feature, or it is disabled. Please make sure that you have a reasonably modern browser (e.g. IE8+, Chrome 4+, Safari 4+, FireFox 3.6+, Opera 10.61+), and that you don't have cross-document messaging disabled in your browser settings.

**Comment by nodiscc** [comment]
> how does one create screenshot of the Request Policy

Your linux distribution should have a screenshot tool installed; I use `xfce4-screenshooter` but this may be `gnome-screenshot` depending on your Desktop Environment. Ca

**Comment by nodiscc** [comment]
> Also once the deny_trackers is set, there doesn't seem to be a way to delete it, is there ?

This is indeed blocked by the "Deny trackers" subscription you subscribed to. There is no way to disable a particular subscription item for now, it's a known issue: https://github.com/RequestPolicyContinued/requestpolicy/issues/491

The `Block       *.ads.imgur.com deny_trackers` rule should not block your browsing since it only block images from Imgur's **advertisement** subdomain.

**Comment by shirishag75** [comment]
>About the Stackexchange login issue, disable RequestPolicy and see if >it fixes it. If yes, please report back.

There seem to be quite a few issues there. 

It turns out that the login issue was failing due to another addon HTTPS Everywhere. Apparently, the stackexchange folks haven't done full https re-direction (whatever the word means) . Once I disabled the two rules for Stackexchange I was able to login to meta.unix.stackexchange without an issue. 

BUT as far as https://meta.stackoverflow.com/network-login-help is concerned, the messaging failure issue is still there. 

    Messaging failed

    Your browser does not support the HTML5 postMessage feature, or it is disabled. Please make sure that you have a reasonably modern browser (e.g. IE8+, Chrome 4+, Safari 4+, FireFox 3.6+, Opera 10.61+), and that you don't have cross-document messaging disabled in your browser settings. 

And I have no idea what this **cross-document messaging** bit is and where it is in my Firefox 36.0 on Debian.

**Comment by nodiscc** [comment]
@shirishag75 Disable RequestPolicy and see if it fixes it. If yes, please report back. If no, this is not a RequestPolicy bug. I don't have this message on Debian, iceweasel 36.0-2.

**Comment by rezad1393** [comment]
it is related to websockets i think.
if you open firefox browser console you will see this error at that page loading
"The connection to wss://qa.sockets.stackexchange.com/ was interrupted while the page was loading."

**Comment by rezad1393** [comment]
first step is to see if you have apps on linux(or your os) that are using port 1900.that is the websocket port in firefox (can see it in about:config ) you can change that port or close the other apps that use that port too.though that didnt solve my problem maybe that helps you.

**Comment by shirishag75** [comment]
@rezad1393 are you talking about https://meta.stackoverflow.com/network-login-help or http://meta.unix.stackexchange.com/ and logging in there ?

@nodiscc both pages and sites seem to be non-requestpolicy bugs.  The first one  https://meta.stackoverflow.com/network-login-help passes all ONLY when both requestpolicy and https-everywhere addons are both disabled. If only RPC (Requestpolicy) is disabled then still I get the messaging failure. It seems the issue is at stackexchange side (see http://meta.stackexchange.com/questions/250620/messaging-failed-doesnt-give-more-info-as-to-what-to-do-about-it#comment818109_250620)

As far as logging in to meta.unix.stackexchange.com is concerned, that was accomplished the moment I disabled stackexchange rules in HTTPS Everywhere.

**Comment by nodiscc** [comment]
Please keep issues not related to RPContinued off the issue tracker...

@shirishag75 If the https://meta.stackoverflow.com/network-login-help problem does not happen with **only** RPContinued **enabled**, it's not a RequestPolicy bug.

**Comment by rezad1393** [comment]
https://meta.stackoverflow.com/network-login-help this fails on html5 postmessage for me even with a new empty firefox profile.
i dont know why.

**Comment by shirishag75** [comment]
@nodiscc understood, thanx for being patient. It is definitely not an RPC bug.

**Comment by myrdd** [comment]
Version `1.0.beta9` is released! This time you still need to restart firefox to update. After that, RequestPolicy will be restartless.

**Comment by myrdd** [comment]
I've created [release `0.5.29`](https://github.com/RequestPolicyContinued/requestpolicy/releases/tag/v0.5.29) and [published it at AMO](https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/) (Mozilla's Add-Ons website). As the `1.0.beta*` releases, it's signed.

This is the first release of the `0.5.*` branch by the RequestPolicyContinued project. For details, see the release notes, copy-pasted below.

------------------------

**Release notes for RequestPolicy 0.5.29:**

* The extension's ID [[1]](https://developer.mozilla.org/en-US/Add-ons/Install_Manifests#id) has been changed.
  The old ID was `requestpolicy@requestpolicy.com`,
  the new one is `rpcontinued@requestpolicy.org`.
  Some details:
  * The main reason is that the old ID has been disabled by Mozilla
    for Firefox 38+ because of a bug [[2]](https://bugzilla.mozilla.org/show_bug.cgi?id=1077185) and because the "old" add-on
    was unmaintained.
  * What does this mean?
    - With the new ID it's again possible to use RequestPolicy 0.5.*
      on Firefox 38 or later.
    - In addition to that, the changed ID allowed us to upload the
      add-on to Mozilla's Add-Ons website (aka AMO). You can find it
      at [[3]](https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/). However, if you're reading this, the add-on might still
      be "unreviewed" [[4]](https://addons.mozilla.org/en-US/faq#unreviewed) and therefore not to be found via search.
* A warning is shown if the extension with the old extension ID is
  installed and enabled. It's not possible to have both add-ons
  enabled in the same browser at the same time.

[1] Info about extension IDs
    https://developer.mozilla.org/en-US/Add-ons/Install_Manifests#id
[2] Mozilla Bug regarding RequestPolicy
    https://bugzilla.mozilla.org/show_bug.cgi?id=1077185
[3] RequestPolicy Continued on Mozilla's Add-Ons website
    https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/
[4] Info about "unreviewed" add-ons
    https://addons.mozilla.org/en-US/faq#unreviewed

**Comment by antnythr** [comment]
I just started using RequestPolicyContinued and noticed that the Ctrl+Alt+R key combo is now associated with the extension.

Is this something that I can change?

**Comment by myrdd** [comment]
Hi @antnythr, that's right. The shortcut is currently not configurable, is that what you want? If yes, check out https://github.com/RequestPolicyContinued/requestpolicy/issues/616.

**Comment by antnythr** [comment]
@myrdd, I had skimmed over some of the issues but missed this one. Thanks.

**Comment by jdgalt** [comment]
I'm not sure what is the right place to post this:

I would like to suggest that we be allowed to assign a priority number to each rule, and have the higher priorities automatically trump the lower ones.

We could allow, for example, priorities from 1 to 255, with any rules that come from a subscription being automatically assigned 128.  That way I can write rules that always yield to them and rules that never do.

Here's an example of a case where this would help.  I use reddit, so have a rule allowing all requests from them.  But I also have a rule banning all requests to *.4disply.com (a site that generates pop-under ads).  Result: One of their pop-under ads appeared, and there is apparently no way to disable those unless I want to take the time to discover and list every single site that reddit is allowed to send requests to.  Setting a high priority for "Block *.4display.com" would solve this problem.

**Comment by myrdd** [comment]
Thanks for sharing your idea @jdgalt. I've already been thinking about adding a "priority" property to each rule. IMHO the default priority could be zero, allowing both positive and negative numbers, but that's a detail we can discuss at a specific issue. It would be nice if you could open a new issue for priorities.

However, the problem in the example case you're describing might have to do with issue #491. Is your default policy "allow"?

**Comment by jdgalt** [comment]
Yes it is, and #498  could be related.

Created #717.

**Comment by myrdd** [comment]
I guess you mean #491 – since your default policy is "allow", that's the fallback policy.

**Comment by woxcab** [comment]
RequestPolicy blocks external images on localhost web-apps (example: web images from non-localhost are not shown in Tiny Tiny RSS (webapp on localhost:8080) when RequestPolicy is enabled). Allowing 'blocked destinations' from list does not help (images are still blocked). Manual policies:

- Origin `localhost`, Destination `*`
- Origin `localhost:8080`, Destination `*`

does not help.

 Can it be fixed manually using policies or is it a bug?

**Comment by myrdd** [comment]
@woxcab Matching against `*` [does not work yet](https://github.com/RequestPolicyContinued/requestpolicy/pull/555). Please try a rule with the "destination" field being empty. If you specify a port, make sure you enter it in the dedicated "port" field. If this doesn't help, please check the Request Log for the exact destination URI.

**Comment by woxcab** [comment]
@myrdd, thank you! Empty destination have helped. For my case origin must be with port `localhost:8080` (if no port then images are blocked).

**Comment by myrdd** [comment]
@woxcab great! By the way, the `*` issue should now be resolved in the [latest pre-build](https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/versions/1.0.beta11.1.1488.r608f5fc.pre). Would you mind trying if `*` now works as the destination host? :)

**Comment by woxcab** [comment]
@myrdd, yes, `*` destination works in the latest pre-build for me. Thank you!

**Comment by agilbertson1977** [comment]
Posting here to start with to make sure that this isn't an open issue (I didn't find it when searching for "font" or "CSS" but maybe I'm not familiar enough with the issue to figure out what the right search term is).

Going to http://www.tampabay.com/news/breaking/ with Request Policy Continued results in links to stories with an associated picture having a weird character that looks like a black circle with a white + in it showing up behind the headline text.  Visiting the same page without Request Policy Continued results in an icon which I guess is supposed to represent a picture of mountains and the sun.  I don't see anything that's not allowed that I would expect to prevent this transformation.

(Let me know if you'd like screenshots of it with/without RPC enabled.)

Is this expected behavior, an existing issue, or should I open a new issue for it?

Thanks!

**Comment by myrdd** [comment]
@agilbertson1977 In this case please create a new issue. Please attach a screenshot of the expected and the actual appearance. Please also add a screenshot of RP's menu, to see which destinations have been allowed/denied.
Right now I couldn't reproduce what you are describing, but that could be because certain rules need to be added, or because of some add-on you've got installed.

**Comment by akwala** [comment]
Recently I've been seeing blocked destinations such as "jid1-mnnxcxisbpnsxq-eff-at-jetpack". So far all of these have ended with "-eff-at-jetpack". My guess is that this has to do with an EFF addon. What is going on here?

**Comment by myrdd** [comment]
@akwala `jid1-MnnxcxisBPnSXQ@jetpack` is the ID of PrivacyBadger ([source](https://github.com/EFForg/privacybadgerfirefox/blob/c3e2031748fd77dd0d96c06637cc227b920ad221/package.json#L4)). How does the request look like, that is, the full URI of origin and destination? It could have to do with the fix for #783, released in `beta12.0`. If you think there is some misbehavior, go ahead and create a new issue.

**Comment by inril** [comment]
Using 1.0.beta12.4.
When enabling the privacy subscription the policy will allow facebook and twitter (7 entries).
I did not expect to see any green entries in a privacy policy. 
Everything in this list should be denied.

Am I missing something?

**Comment by myrdd** [comment]
> When enabling the privacy subscription the policy will allow facebook and twitter (7 entries).

These seven entries allow facebook-internal and twitter-internal request, respectively. There are also blocking rules for those domains. So, for example, requests to `facebook.net` are blocked, except requests from `facebook.com`. See this image for clarification:

![deny_trackers rules for facebook](https://cloud.githubusercontent.com/assets/4286829/20815746/734ff37a-b81f-11e6-96df-a8d967434239.jpg)

**Comment by Atavic** [comment]
Without a minimal Allow Policy I can't even do a complete search on twitter.

**Comment by nodiscc** [comment]
Hi @myrdd , it's been a long time, hope all is well.
I have sent a PR at https://github.com/RequestPolicyContinued/subscriptions/pull/54 I wish you could review (serials, etc.).
I want to remove the deny_trackers subscription for reasons I will detail in another PR.

@Atavic in "default deny" mode you need to whitelist items manually to get full website functionality, that's the point.
If you need some basic premade allow rules, enable subscriptions.
If you want to add allow rules to official subscriptions, check https://github.com/RequestPolicyContinued/subscriptions

**Comment by lazerhawk** [comment]
Mozilla's add-on compatibility reporter shows it isn't multiprocess/e10s compatible. Reporter bug or is it actually true?

**Comment by Atavic** [comment]
According to: https://www.arewee10syet.com/
 RequestPolicy Continued • incompatible

**Comment by myrdd** [comment]
> Mozilla's add-on compatibility reporter shows it isn't multiprocess/e10s compatible. Reporter bug or is it actually true?

RP is not yet e10s compatible. See
https://github.com/RequestPolicyContinued/requestpolicy/issues/828#issuecomment-280920383

**Comment by nazmifr** [comment]
hey, I have some kind of suggestion for that amazing piece of addon that is requestpolicy:

When there are some redirects, the user has to move his mouse all the way to the upper right hand corner of Firefox to click on _allow_, it's painful and sometimes there are timeouts (especially if you're slow or on a laptop), couldn't an enter key press fill the same function while being simpler to acheive?

Thanks for having read
Nazmi

**Comment by myrdd** [comment]
Thanks for your comment @nazmifr. If you like to use the keyboard, you currently can press Alt+A to allow the request. If it's not „A“, it's the underlined letter on the „allow“ button.

**Comment by nazmifr** [comment]
Thanks for your response, I will from now on.

**Comment by zazenbingle** [comment]
Hello, 

Please help me configure an "Allow" rule for web browsing.  I perform a search on DuckDuckGo, then right-click some results and open in new tab.  When I look at each "new tab", the website is blocked.  

I created the following rule: 
Policy --	Origin----------------------- 	Destination--	Rule Set 	
Allow  --	https://duckduckgo.com/://*:*	*://*:*	 -- 	User

What am I doing wrong?

**Comment by myrdd** [comment]
> I created the following rule:
> Policy -- Origin----------------------- Destination-- Rule Set
> Allow -- https://duckduckgo.com/://:	://:* -- User

I guess you put `https://duckduckgo.com/` in the origin "scheme" field and `://` in the destination "scheme" field. Instead, fill in as follows:

* origin scheme: `https`
* origin host: `duckduckgo.com`
* destination host: `*`

**Comment by zazenbingle** [comment]
Thank you!  It worked! 

I'm thankful for this add-on and for your responsiveness.  

Here are my 2 cents on how to make it more noob-friendly: 

1. Change default setting: 
--> Preferences >> Menu >> "Allow adding non-temporary rules when using Private Browsing Mode" 
--> Current default: OFF
--> Proposed default: ON 
--> Reason: Security conscious noobs will have their browsers set to private mode or equivalent.  
--> When I didn't see the option to whitelist my search engine website, it just confused me until I found this setting.  

2. Modify the "Allow requests from *.duckduckgo.com" [or whatever website you're on] command: 

--> Current "Allow" rule created by the command: 
Policy - Origin - Destination - Rule Set
Allow - https://duckduckgo.com/://:  -  ://:* - User

--> Proposed "Allow" rule created by the command: 
Policy -	Origin			- Destination 	- Rule Set 	
Allow -	https://duckduckgo.com	- *		- User	

--> Reason for the change: the current automated "Allow Rule" creation didn't work for me.  I scoured the Internet for a couple of hours before creating a GitHub account and posting this question.  Not complaining, just providing the perspective of a motivated noob.    

3. Add an FAQ explaining how to configure a search engine website in this way.  
--> If changing the default setting is not acceptable, then you can just include those instructions in the FAQ.    
--> If changing the values created by the "Allow requests from..." command is time-consuming, then posting this information in an FAQ will be very helpful to prevent future questions like mine.  
--> This is probably the first rule a user will create, right?  I mean, don't many people browse by opening new tabs? 
--> If my assumptions are incorrect, then how are they incorrect?  Maybe it's a completely different FAQ that's required, and I just don't know enough to understand it.

**Comment by zazenbingle** [comment]
myrdd,

Scratch my last post.  It seems more likely that I didn't see the setting until AFTER I created it manually, because the "Allow requests from" command seems to work everywhere else.  

I still think an FAQ that helps a new user get their search page up and running is a good idea.  Here's what I propose: 

FAQ: My search engine web-page can't open any pages without me clicking on a link to "Allow" on each new page I open. What should I do?  

ANS: Here's how you add a rule to allow your search engine web-page to open new pages: 
1. From your search enging web-page, hover over the red-flag icon for the "RequestPolicy Continued" extension.  
2. Click on "Allow requests from *.duckduckgo.com" [or whatever website you perform web-searches from]

If you don't see the option to "Allow requests from" option when you hover over the icon: 
1. You're probably in private-mode or equivalent.  
2. Go to: Settings >> Preferences >> Menu >> "Allow adding non-temporary rules when using Private Browsing Mode" 
3. Change from unchecked to checked
4. You should now see the option to "Allow requests from..." when you hover over the icon.

**Comment by myrdd** [comment]
Hi @zazenbingle,

> FAQ: My search engine web-page can't open any pages without me clicking on a link to "Allow" on each new page I open. What should I do?  
> 
> ANS: Here's how you add a rule to allow your search engine web-page to open new pages: 
> 1. From your search enging web-page, hover over the red-flag icon for the "RequestPolicy Continued" extension.  
> 2. Click on "Allow requests from *.duckduckgo.com" [or whatever website you perform web-searches from]

Actually you should get a redirection notification when clicking on the link on duckduckgo.com. See here: https://github.com/RequestPolicyContinued/requestpolicy/issues/859

**Comment by rain0r** [comment]
Are there any plans on migrating the add-on to Firefox 57.0 ("Firefox Quantum")?

**Comment by jrrdev** [comment]
Work in progress, see issue #704 :smiley:

**Comment by TPS** [comment]
I've recently received notification of a new release, but there's only an unbuilt version.  Since AMO's dropped betas, where does 1 go to get prebuilt 1s?

**Comment by myrdd** [comment]
@TPS I put a link to the installation URL just now in the first post of #754.
It's https://sslsites.de/requestpolicy.256k.de/requestpolicy-nightly.xpi
Make sure to disable any other RP/RPC versions before installation.

**Comment by TPS** [comment]
@myrdd Thanks very much! :bow:

**Comment by vertigo220** [comment]
Am I missing something, or is there no way to export/import the settings and custom rules? Also, I'm confused with regard to the subscriptions, since I would expect major sites like Amazon and Newegg to work with them, but I had to manually allow things to get them to not be complete disasters.

Another thing, the interface is a bit confusing. You can select multiple things at once (e.g. you can select/underline permanently and temporarily allow and block, all at the same time) and it's not very clear what's going on. And the meaning of the (x+y) numbers isn't readily apparent (it's not difficult to figure out, but it would be nice if there was a key or something). It's also not clear what the different icons (?, do not enter sign (red circle with white dash), etc) or "mixed" destinations mean.

And finally, the refresh page after whitelisting option doesn't work for me. I may create a separate issue for that, but it'll have to wait until I do more troubleshooting, so I figured I'd just mention it here for now. Frankly, I'm not sure RPC provides much, if anything, beyond what uBo and uMatrix do for me, and so I don't know if I'll continue using it, but it does seem like a good project, and I'd like to see it continue to improve, if for no other reason than to have options.

**Comment by ldgbc** [comment]
@vertigo220 Just use uMatrix it is a good alternative and RP is dead at the moment.

**Comment by shirishag75** [comment]
for the record, he just made a new release, although I have yet to get it get it compiled properly https://github.com/RequestPolicyContinued/requestpolicy/issues/907 and then see if it is able to install it in 60.08.0esr although soon it will be 68.0.x esr according to mozilla upstream.
